applications/flash_attention_v2/collective/xe_flash_attn_mma.hpp

```diff
@@ -300,8 +302,54 @@ struct CollectiveMmaAttention<MainloopIntelPVC<Stages>, TileShape_, ElementQ_, S
     copy(params.gmem_tiled_copy_v, tVgV, tVrV);
     cute::gemm(tiled_mma, accum, tPr, tCrV, frag_src);
   }
+
+  template <bool VarLen, class ProblemShape>
+  CUTLASS_DEVICE static constexpr Params get_updated_copies(Params const& params, ProblemShape const& problem_shape, int const& l_coord) {
+    if constexpr (!VarLen) {
+      return params;
+    } else {
+      auto [batch, num_heads, seq_len_qo, seq_len_kv, head_size_qk, head_size_vo] = problem_shape;
+
+      auto qo_cumulative_length = get<2>(problem_shape).cumulative_length;
+      auto kv_cumulative_length = get<3>(problem_shape).cumulative_length;
+
+      int offset_q = l_coord == 0 ? 0 : num_heads * head_size_qk * qo_cumulative_length[l_coord];
+      int offset_k = l_coord == 0 ? 0 : num_heads * head_size_qk * kv_cumulative_length[l_coord];
+      int offset_v = l_coord == 0 ? 0 : num_heads * head_size_vo * kv_cumulative_length[l_coord];
+
+      auto q_traits = static_cast<traits_load_Q const&>(params.gmem_tiled_copy_q);
+      ElementQ* q_ptr = (ElementQ*)q_traits.base_ptr;
+
+      auto k_traits = static_cast<traits_load_K const&>(params.gmem_tiled_copy_k);
+      ElementK* k_ptr = (ElementK*)k_traits.base_ptr;
+
+      auto v_traits = static_cast<traits_load_V const&>(params.gmem_tiled_copy_v);
+      ElementV* v_ptr = (ElementV*)v_traits.base_ptr;
+
+      auto tensorQ = make_tensor(make_gmem_ptr(q_ptr + offset_q),
+                                make_layout(make_shape(seq_len_qo, head_size_qk, num_heads),
+                                            make_stride(static_cast<int64_t>(head_size_qk), cute::C<1>{}, static_cast<int64_t>(seq_len_qo * head_size_qk))));
+      auto tensorK = make_tensor(make_gmem_ptr(k_ptr + offset_k),
+                                make_layout(make_shape(seq_len_kv, head_size_qk, num_heads),
+                                            make_stride(static_cast<int64_t>(head_size_qk), cute::C<1>{}, static_cast<int64_t>(seq_len_kv * head_size_qk))));
+      auto tensorV = make_tensor(make_gmem_ptr(v_ptr + offset_v),
+                                make_layout(make_shape(head_size_vo, seq_len_kv, num_heads),
+                                            make_stride(cute::C<1>{}, static_cast<int64_t>(head_size_vo), static_cast<int64_t>(seq_len_kv * head_size_vo))));
+      XE_Copy_Q copyQ = make_tiled_copy(atom_load_Q{}.with(tensorQ),
+                                        Layout<CopyThreadShape>{},
+                                        make_layout(shape_div(typename traits_load_Q::BlockShape{}, CopyThreadShape{})));
```

Comment:
```suggestion
      XE_Copy_Q copyQ = XE_Copy_Q{}.with(tensorQ);
```

and the same for other inputs

Reply:
I think it needs to be something like:
```cpp
 XE_Copy_Q copyQ{XE_Copy_Q{}.with(tensorQ)};
```

because `with()` returns traits as opposed to a tiled copy.

Reply:
Yes you are correct